packages/plugins/Traderxyz/src/SNSAdaptor/components/NftListView.tsx

```diff
@@ -0,0 +1,146 @@
+        },
+    }))
+
+    const CustomTreeChild = styled(
+        (props: TreeItemProps & { ContentProps?: { collectionImage: string | undefined } }) => (
+            <TreeItem {...props} ContentComponent={TreeChildContent} />
+        ),
+    )(({ theme }) => ({
+        [`& .${treeItemClasses.iconContainer}`]: {
+            '& .close': {
+                opacity: 0.3,
+            },
+        },
+        [`& .${treeItemClasses.group}`]: {
+            marginLeft: 15,
+            paddingLeft: 18,
+            borderLeft: `1px solid ${alpha(theme.palette.text.primary, 0.4)}`,
+        },
+    }))
+
+    const CustomTreeChildItem = styled(
+        (
+            props: TreeItemProps & {
+                ContentProps?: {
+                    nftData?: TreeNftData | undefined
+                    handleSelection?(collection_index: number, item_index: number, type?: string): void
+                }
+            },
+        ) => <TreeItem {...props} className={props.className} ContentComponent={TreeChildItemContent} />,
+    )(({ theme }) => ({
+        [`& .${treeItemClasses.iconContainer}`]: {
+            '& .close': {
+                opacity: 0.3,
+            },
+        },
+        [`& .${treeItemClasses.group}`]: {
+            marginLeft: 15,
+            paddingLeft: 18,
+            borderLeft: `1px solid ${alpha(theme.palette.text.primary, 0.4)}`,
+        },
+    }))
+
+    const collectionItemList = props?.nftList?.map((item, index) => {
+        const collectionImage = item.tokens[0].image_preview_url
+        return (
+            <CustomTreeChild
+                nodeId={'p-' + index}
+                key={'p-' + index}
+                classes={{ label: classes.label }}
+                label={item.collection_name}
+                ContentProps={{ collectionImage: collectionImage }}>
+                {item?.tokens?.map((item, i) => (
+                    <CustomTreeChildItem
+                        nodeId={'p-' + index + 'c-' + i}
+                        key={'p-' + index + 'c-' + i}
+                        label={item.name}
+                        ContentProps={{
+                            nftData: {
+                                id: item.id,
+                                image_preview_url: item.image_preview_url,
+                                token_id: item.token_id,
+                                is_selected: item.is_selected,
+                                collection_index: index,
+                                item_index: i,
+                            },
+                            handleSelection: props.handleSelection,
+                        }}
+                    />
+                ))}
+            </CustomTreeChild>
+        )
+    })
+
+    const previewImages1 = props?.nftList
+        ?.map((x) => x?.tokens)
+        .flat()
+        .map((y) => {
+            return y?.image_preview_url
+        })
+        .slice(0, 3)
```

Comment:
slice then map.
Better to memoize it with useMemo
```suggestion
    const previewImages1 = props?.nftList
        ?.map((x) => x?.tokens)
        .flat()
        .slice(0, 3)
        .map((y) =>  y?.image_preview_url)
```

Reply:
Fixed